### PR TITLE
3.8

### PR DESCRIPTION
**Pull request by mingwandroid**
Checklist
* [x] Used a fork of the feedstock to propose changes
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,40 +1,36 @@
...
+{% set version = "3.8.0b3" %}
 {% set linkage_nature = os.environ.get('PY_INTERP_LINKAGE_NATURE', '') %}
 {% if linkage_nature != '' %}
   {% set linkage_nature = "_" ~ linkage_nature %}
 {% endif %}
 
 package:
   name: python
-  version: {{ version }}
+  version: {{ base_version }}
```
Comment (jjhelmus) [inline recipe/meta.yaml]: I am opposed to faking the version here.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -63,62 +59,75 @@ source:
     folder: externals/nasm-2.11.06                                                   # [win]
     sha256: de3c87b26a80e789986d8e6950c6304175d3829afe9c6c7211eb7257266ab0ac         # [win]
 
+
 build:
-  number: 1
+  number: 0
+  add_pip_as_python_dependency: false
```
Comment (jjhelmus) [inline recipe/meta.yaml]: disable_pip?

File: recipe/meta.yaml [hunk 3]
```diff
@@ -144,4 +153,3 @@ extra:
     - msarahan
     - pelson
     - ocefpaf
-    - scopatz
```
Comment (jjhelmus) [inline recipe/meta.yaml]: should be retained

File: recipe/sysconfigdata/_sysconfigdata_aarch64_conda_cos7_linux_gnu.py [hunk 4]
```diff
@@ -1,805 +0,0 @@
...
- 'WITH_PYMALLOC': 1,
- 'WITH_VALGRIND': 0,
- 'X87_DOUBLE_ROUNDING': 0,
- 'XMLLIBSUBDIRS': 'xml xml/dom xml/etree xml/parsers xml/sax',
- 'abs_builddir': '/opt/conda/conda-bld/python_1512347718438/work/build-static',
- 'abs_srcdir': '/opt/conda/conda-bld/python_1512347718438/work',
- 'datarootdir': '/opt/anaconda1anaconda2anaconda3/share',
- 'exec_prefix': '/opt/anaconda1anaconda2anaconda3',
- 'prefix': '/opt/anaconda1anaconda2anaconda3',
- 'srcdir': '/opt/conda/conda-bld/python_1512347718438/work'}
```
Comment (jjhelmus) [inline recipe/sysconfigdata/_sysconfigdata_aarch64_conda_cos7_linux_gnu.py]: Needs to be retained

File: recipe/sysconfigdata/default/_sysconfigdata_linux.py [hunk 5]
```diff
@@ -684,9 +684,24 @@
...
+if GCC:
+    with open(devnull, 'w') as err:
+        GCC_VERSION = check_output([GCC, '--version'], stderr=err).decode('utf-8').splitlines()[0].split(' ')[-1]
+    print(GCC_VERSION)
+    for btv in ('BLDSHARED', 'CC', 'CXX', 'LDCXXSHARED',
+                'LDSHARED', 'LINKCC', 'MAINCC'):
+        if btv in build_time_vars:
+            build_time_vars[btv] = build_time_vars[btv]. \
+                replace('-B /opt/anaconda1anaconda2anaconda3/compiler_compat ',
+                        '')
```
Comment (jjhelmus) [inline recipe/sysconfigdata/default/_sysconfigdata_linux.py]: Should this be added to the other linux platforms?

File: recipe/meta.yaml [hunk 6]
```diff
@@ -63,62 +59,75 @@ source:
...
+   # # conda-build bug?
+   #  - ncurses >=6.1,<7.0a0               # [unix]
 
 test:
+  # Because we need to use xvfb-run on linux, from a CDT package.
+  requires:  # [linux]
+    # We are running centos6 software here (on whatever distro you are on)
+    # .. so it best be compatible (you should use centos6 if possible).
+    - {{ cdt('xorg-x11-server-xvfb') }}  # [linux]
+    - {{ cdt('xorg-x11-xauth') }}  # [linux]
```
Comment (jjhelmus) [inline recipe/meta.yaml]: These CDT do not exist yet.  They will need to be created.

File: recipe/build.sh [hunk 7]
```diff
@@ -27,20 +95,14 @@ if [[ ${PY_INTERP_LINKAGE_NATURE} == shared ]]; then
   _ENABLE_SHARED=--enable-shared
 fi
 
-# For debugging builds, set this to 0 to disable profile-guided optimization
+# For debugging builds, set this to no to disable profile-guided optimization
```
Comment (jjhelmus) [inline recipe/build.sh]: Is this still true or does QUICK_BUILD take precedence

File: recipe/meta.yaml [hunk 8]
```diff
@@ -63,62 +59,75 @@ source:
...
-  number: 1
+  number: 0
+  add_pip_as_python_dependency: false
   # Windows has issues updating python if conda is using files itself.
   # Copy rather than link.
   no_link:
     - DLLs/_ctypes.pyd  # [win]
-    - bin/python3.7  # [not win]
-  script_env:
-    - python_branding
```
Comment (jjhelmus) [inline recipe/meta.yaml]: I think this is still needed

File: recipe/bld.bat [hunk 9]
```diff
@@ -1,10 +1,10 @@
 setlocal EnableDelayedExpansion
 
-:: brand Python with conda-forge startup message
-%SYS_PYTHON% %RECIPE_DIR%\brand_python.py
-if errorlevel 1 exit 1
+REM brand Python with conda-forge startup message
+REM %SYS_PYTHON% %RECIPE_DIR%\brand_python.py
+REM if errorlevel 1 exit 1
```
Comment (jjhelmus) [inline recipe/bld.bat]: This is needed for conda-forge, unless a patch added similar messaging.

**Comment by jjhelmus** [comment]
The CI is failing because the new CDTs do not exist.  

@mingwandroid Could you target this against the `pre-3.8` branch rather than master?

File: recipe/ld.wrapper [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+       [[ ${GCC_SYSV} =~ 7[.].* ]]; then
+      $(dirname "${BASH_SOURCE[0]}")/ld.bin "$@"
+    else
+      "${_GCC}" "$@"
+    fi
+  fi
+}
+
+_COMPILER_COMPAT_LD_ARGS=( "$@" )
+ld_fn "${_COMPILER_COMPAT_LD_ARGS[@]}"
```
[the remark this answers is not in the document]
Comment (mingwandroid) [reply recipe/ld.wrapper]: Just a left over.

File: recipe/sysconfigdata/default/_sysconfigdata_linux.py (hunk 5 above)
Comment (mingwandroid) [reply recipe/sysconfigdata/default/_sysconfigdata_linux.py]: This is used on all linux platforms when our compilers are not in use is it not? Should be?!

File: recipe/meta.yaml (hunk 6 above)
Comment (mingwandroid) [reply recipe/meta.yaml]: Just `xorg-x11-xauth`. I lost my phone's SIM unlock code (yeah really) so haven't been able to run `c3i` to build it yet. It's ok, I recovered it now.

File: recipe/build.sh (hunk 7 above)
Comment (mingwandroid) [reply recipe/build.sh]: Not sure, will get cleaned up.

File: recipe/meta.yaml (hunk 8 above)
Comment (mingwandroid) [reply recipe/meta.yaml]: But not @3.7 right?

File: recipe/bld.bat (hunk 9 above)
Comment (mingwandroid) [reply recipe/bld.bat]: Yeah, how can we try to keep our recipes in sync. What do you think about using env vars like I do for static vs dynamic Python in meta.yaml? I'm OK with doing something like that personally.

File: recipe/meta.yaml (hunk 3 above)
Comment (mingwandroid) [reply recipe/meta.yaml]: Indeed, sorry.

**Comment by mingwandroid** [comment]
I can indeed target it there. @jjhelmus, do you want to take this over for now? We need version checks for newer and probably some env var overrides. If you want to address only this part independently of the other changes I'm making (which will get more invasive until I have it cleaned up), then feel free to take what I've done in `recipe/sysconfigdata/default/_sysconfigdata_linux.py` and make a separate PR from it to conda-forge and I'll rebase on top next week.

File: recipe/meta.yaml (hunk 1 above)
Comment (mingwandroid) [reply recipe/meta.yaml]: OK, please advise on better suggestions for betas with version comparison considerations, err, considered (carefullly!).

File: recipe/bld.bat (hunk 9 above)
Comment (mariusvniekerk) [reply recipe/bld.bat]: Sounds cleaner

**Comment by mariusvniekerk** [comment]
closed by #274

**Pull request closed without merging**